# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/public/experimental/TypedData.h
# Commit: e162d996deac
# Full Hash: e162d996deac639a030daa2c3c156cbdb2d442b4
# Author: Steve Fink <sfink@mozilla.com>
# Date: 2021-09-09 15:22:41
# Regressor Bug: 1720422
# File Overlap Count: 1
# Description:
#   Bug 1720422 - Switch DOM bindings typed arrays to using JSAPI templates r=jonco,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D119858
# ==============================================================================

diff -r 5445f480023e -r e162d996deac js/public/experimental/TypedData.h
--- a/js/public/experimental/TypedData.h	Wed Sep 08 23:47:00 2021 +0000
+++ b/js/public/experimental/TypedData.h	Wed Sep 08 23:47:00 2021 +0000
@@ -299,6 +299,11 @@
 // ArrayBufferObject or some sort of ArrayBufferViewObject (see below).
 // Note that this will always hold an unwrapped object.
 class JS_PUBLIC_API ArrayBufferOrView {
+ public:
+  // Typed Arrays will set this to their specific element type.
+  // Everything else just claims to expose things as uint8_t*.
+  using DataType = uint8_t;
+
  protected:
   Heap<JSObject*> obj;
 
